config-provisioning/src/main/java/com/yahoo/config/provision/WireguardKey.java

```diff
@@ -0,0 +1,30 @@
+package com.yahoo.config.provision;
+
+import ai.vespa.validation.PatternedStringWrapper;
+
+import java.util.regex.Pattern;
+
+/**
+ * Wraps a Wireguard key.
+ * For security reasons, this should only be used for public keys, although private keys use the same format.
+ *
+ * @author gjoranv
+ */
+public class WireguardKey extends PatternedStringWrapper<WireguardKey> {
+
+    // See https://lists.zx2c4.com/pipermail/wireguard/2020-December/006222.html
+    private static final Pattern pattern = Pattern.compile("^[A-Za-z0-9+/]{42}[A|E|I|M|Q|U|Y|c|g|k|o|s|w|4|8|0]=$");
```

Comment:
Square brackets define a list of characters of which 1 should be matched, so the `|` are redundant and as it is now, it'll also allow a `|` characters in second last spot
```suggestion
    private static final Pattern pattern = Pattern.compile("^[A-Za-z0-9+/]{42}[AEIMQUYcgkosw480]=$");
```